### PR TITLE
Code changes to meet DB changes. Also added a simple email template f…

### PR DESCRIPTION
**Pull request by bibinss**
…or alert emails

### CODE REVIEW

File: src/main/resources/templates/emailTemplate.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+<p>
+    You have one alert generated by detector "<span th:text="${detector}"></span>" for metric with key "<span th:text="${metricKey}"></span>"
+    and value "<span th:text="${metricValue}"></span>".
+</p>
+<p>More details on the alert message are:</p>
+<ul>
+    <li th:each="tag : ${metadata}" th:text="${tag}"></li>
+</ul>
+
+<p>Alert subscription rules can be access from <a href="">console</a>.</p>
```
Comment (ghost) [inline src/main/resources/templates/emailTemplate.html]: Spelling error: "access" should be "accessed" in this case.

Comment (williewheeler) [reply src/main/resources/templates/emailTemplate.html]: Even better, use an active voice:

"You can access alert subscription rules from the <a href="...">console</a>."

Comment (ghost) [reply src/main/resources/templates/emailTemplate.html]: +1 I just didn't want to get into a grammar and usage discussion when there's coding to be done 😄

Comment (bibinss) [reply src/main/resources/templates/emailTemplate.html]: done

File: src/main/java/com/expedia/alertmanager/dao/SubscriptionRepository.java [hunk 2]
```diff
@@ -16,8 +16,12 @@
...
 
 import com.expedia.alertmanager.entity.Subscription;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 
-public interface SubscriptionRepository extends CrudRepository<Subscription, Long> {
+import java.util.List;
 
+public interface SubscriptionRepository extends CrudRepository<Subscription, Long> {
+    @Query("SELECT s FROM subscription s where s.detectorId = ?1 AND s.metricId = ?2")
```
Comment (williewheeler) [inline src/main/java/com/expedia/alertmanager/dao/SubscriptionRepository.java:25]: You shouldn't need this query. Spring Data automatically derives the query from the method name, as long as you use the canonical format. (Which you are doing here.)

File: src/main/java/com/expedia/alertmanager/entity/Subscription.java [hunk 3]
```diff
@@ -40,17 +40,37 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
+    @Column(name = "metric_id")
```
Comment (williewheeler) [inline src/main/java/com/expedia/alertmanager/entity/Subscription.java:43]: FYI there is a way to make this translation automatic too. It is a one-line JPA config. Let me know if you are interested.

File: src/main/java/com/expedia/alertmanager/entity/Subscription.java [hunk 4]
```diff
@@ -40,17 +40,37 @@
...
     private boolean enabled;
 
+    private String owner;
+
     @Column(name = "date_created")
     private Timestamp timestamp;
 
-    public Subscription(String type, String endpoint) {
+    public Subscription(String metricId, String detectorId, String name, String description,
+                        String type, String endpoint, String owner) {
```
Comment (williewheeler) [inline src/main/java/com/expedia/alertmanager/entity/Subscription.java:67]: Not an awesome constructor--lots of params and they're all Strings. Can we clean this up at all? Maybe builder pattern? (Not a blocker--just something to consider for the future.)

**Review by williewheeler — APPROVED** [review]

**Pull request merged**